dataland-frontend/src/components/general/apiKey/ApiKeyCard.vue

```diff
@@ -68,17 +64,26 @@ export default defineComponent({
   }),
   computed: {
     expiryDateInDays() {
-      return calculateDaysFromNow(this.expiryDate);
+      return calculateDaysFromNow(this.expiryDate * 1000);
     },
 
-    isKeyExpired() {
-      return this.expiryDate >= Date.now();
+    whenKeyExpire() {
+      if (this.expiryDate && this.expiryDate * 1000 >= Date.now()) {
+        return `The API Key will expire on ${formatExpiryDate(this.expiryDateInDays)}`;
+      } else if (this.expiryDate && this.expiryDate * 1000 < Date.now()) {
+        return `The API Key expired ${formatExpiryDate(this.expiryDateInDays)}`;
+      } else {
+        return "The API Key has no defined expire date";
+      }
     },
   },
   methods: {
     deleteConfirmToggle() {
       this.viewDeleteConfirmation = !this.viewDeleteConfirmation;
     },
+    isKeyExpired() {
+      return this.expiryDate * 1000 >= new Date().getTime() || this.expiryDate == null;
+    },
```

Comment:
This method returns "true" if the api key is not expired.
It should be the other way round.

Reply:
ok

Reply:
You could simplify this a little bit by writing
```typescript
isKeyExpired() {
    return this.expiryDate * 1000 < new Date().getTime();
}
```